zp-relayer/workers/sentTxWorker.ts

```diff
@@ -111,16 +108,23 @@ async function handleReverted(
     if (!wj?.id) continue
     waitingJobIds.push(wj.id)
 
-    const { txPayload, traceId } = wj.data
-    const transactions = [txPayload]
+    const { txPayload } = wj.data
+    let reschedulePromise: Promise<any>
 
-    // To not mess up traceId we add each transaction separately
-    const reschedulePromise = poolTxQueue.add(txHash, { transactions, traceId }).then(j => {
-      const newPoolJobId = j.id as string
-      newPoolJobIdMapping[wj.data.poolJobId] = newPoolJobId
-      return newPoolJobId
+    reschedulePromise = poolTxQueue.add(txHash, {
+      type: txPayload.type,
+      transactions: [txPayload.transactions],
```

Comment:
Should it be this way?
```suggestion
      transactions: txPayload.transactions,
```
Also seems like we can just pass the original txPayload object?

Reply:
`poolTxQueue` accepts `BatchTx` with `transactions` being an array. This is primarily done for backward compatibility in case we will want to add some logic where client will send multiple txs in one request.
https://github.com/zkBob/zeropool-relayer/blob/18c0164a4caa1a9f76a4f4c93cfe85bd6a34a4b6/zp-relayer/queue/poolTxQueue.ts#L50-L54

Reply:
But `transactions` is already an array, isn't it?

Reply:
`poolTxQueue` uses type `BatchTx<WorkerTxType, true>`
`sentTxQueue` uses type `BatchTx<WorkerTxType, false>`

So, in `sentTxQueue` job `transactions` is just a `WorkerTx<T>`, not a `WorkerTx<T>[]`.
Agree, that it may be misleading at a first glance.

Reply:
A little more details:
When `poolTxQueue` accepts an array `WorkerTx<T>[]` of length `N` it then submits `N` `sentTxQueue` jobs